### PR TITLE
[Dy2St] Delete legacy class TracedLayer and its related unit tests

### PR DESCRIPTION
**Pull request by Fridge003**
### PR types
Others

### PR changes
Others

### Description

Delete class `TracedLayer` from `paddle/jit/api.py`, as well as several helper methods that only this class calls.

The deleted helper methods include `_trace`, `extract_vars`, `_extract_vars`, `create_program_from_desc`, and they are all located in `paddle/jit/api.py`.

Unit tests related to this class are also deleted.

PCard-66972

### CODE REVIEW

File: test/legacy_test/test_traced_layer_err_msg.py
Comment (SigureMo) [inline test/legacy_test/test_traced_layer_err_msg.py:1]: 这整个文件可以删了

File: test/legacy_test/test_imperative_trace_non_persistable_inputs.py
Comment (SigureMo) [inline test/legacy_test/test_imperative_trace_non_persistable_inputs.py:1]: 这整个文件可以删了

File: python/paddle/jit/api.py [hunk 1]
```diff
@@ -93,26 +87,6 @@ def create_program_from_desc(program_desc):
     return program
```
Comment (SigureMo) [inline python/paddle/jit/api.py]: 这个 `create_program_from_desc` 是不是也可以删了？

File: python/paddle/jit/api.py [hunk 2]
```diff
@@ -1524,380 +1498,6 @@ def load(path, **configs):
...
-@dygraph_only
-def _trace(
-    layer, inputs, feed_prefix='feed_', fetch_prefix='fetch_', tmp_prefix='t_'
-):
-    assert isinstance(layer, Layer)
-
-    if not isinstance(inputs, (list, tuple)):
-        inputs = [inputs]
-
-    tracer = _dygraph_tracer()._get_program_desc_tracer()
```
Comment (SigureMo) [inline python/paddle/jit/api.py:1536]: TODO：下个 PR，着手从 pybind 侧删除 `Tracer._get_program_desc_tracer` 代码

File: test/legacy_test/test_traced_layer_err_msg.py
Comment (Fridge003) [reply test/legacy_test/test_traced_layer_err_msg.py:1]: Done

File: test/legacy_test/test_imperative_trace_non_persistable_inputs.py
Comment (Fridge003) [reply test/legacy_test/test_imperative_trace_non_persistable_inputs.py:1]: Done

File: python/paddle/jit/api.py (hunk 1 above)
Comment (Fridge003) [reply python/paddle/jit/api.py]: Deleted

File: python/paddle/jit/api.py (hunk 2 above)
Comment (Fridge003) [reply python/paddle/jit/api.py:1536]: OK

**Review by SigureMo — APPROVED** [review]
LGTMeow 

目测没啥问题，等 CI 过完应该就可以了～

**Review by XieYunshen — APPROVED** [review]

**Pull request merged**